tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/ConstAndDefRewriter.scala

```diff
@@ -28,7 +28,16 @@ class ConstAndDefRewriter(tracker: TransformationTracker) extends TlaModuleTrans
           logger.error("  > If you need support for n-ary CONSTANTS, write a feature request.")
           throw new OverridingError(msg, overridingDef.body)
         } else if (d.typeTag != overridingDef.body.typeTag) {
-          val msg = s"The types of ${d.name} and ${overridingDef.name} do not match."
+          val dTypeString = d.typeTag match {
+            case Typed(tt) => tt.toString
+            case Untyped() => d.typeTag.toString
+          };
+          val overridingDefTypeString = overridingDef.body.typeTag match {
```

Comment:
I ended up using match statements here to unwrap the types in the type tag (otherwise the output would look like `Typed(Set(Str))` which is more confusing to me than `Set(Str)`. Is there a better way of doing this? I saw that `asTlaType1()` is used elsewhere unwrap type tags, but I'm not confident that `transformDef()` will only ever be called with TlaType1 types.